src/gpu/generic/sycl/ref_binary.hpp

```diff
@@ -50,6 +50,10 @@ struct ref_binary_t : public gpu::generic::sycl::primitive_t {
             const bool ok = set_default_params() == status::success
                     && attr_.set_default_formats(dst_md()) == status::success
                     && check_data_types(src0_d, src1_d, dst_d)
+                    && IMPLICATION(attr()->post_ops_.len() > 0,
+                            src0_d.data_type() != bf16
+                                    && src1_d.data_type() != bf16
+                                    && dst_d.data_type() != u8)
```

Comment:
What is the point of this check? If we have post ops, we cant have bf16 for either src or u8 for dst? Why?

Reply:
Because those cases are specifically skipped in the **ocl** implementation, additionally, checking [this file](https://github.com/oneapi-src/oneDNN/blob/kala855/binary-bf16/tests/benchdnn/binary/binary.cpp#L106:L115) I found that function which is launched when we return a false from the **init** function.

Reply:
If those cases work with SYCL implementation they do not need to be skipped, even if ocl impl skips them.

Reply:
I will check them and if they pass I will enable them.

Reply:
I included the cases and they are passing as expected @t4c1. Thanks